backends/tfhe-cuda-backend/cuda/src/integer/multiplication.cuh

```diff
@@ -248,6 +249,40 @@ __host__ void host_integer_sum_ciphertexts_vec_kb(
 
   auto max_shared_memory = cuda_get_max_shared_memory(gpu_indexes[0]);
 
+  // create lut object for message and carry
+  // we allocate luts_message_carry in the host function (instead of scratch)
+  // to reduce average memory consumption
+  bool release_reused_lut = false;
+  if (reused_lut == nullptr) {
+    release_reused_lut = true;
+    size_t ch_amount = r / chunk_size;
+    if (!ch_amount)
+      ch_amount++;
+    reused_lut = new int_radix_lut<Torus>(streams, gpu_indexes, gpu_count,
+                                          mem_ptr->params, 2,
+                                          2 * ch_amount * num_blocks, true);
+  }
+  int_radix_lut<Torus> *luts_message_carry = reused_lut;
```

Comment:
can we somehow check here that reused_lut has big enough pbs_buffer?

Reply:
I've made a change to do this, now we only reuse the lut if it meets requirements.

Reply:
Hmm doesn't look so easy, let me roll back the change.